crates/client/eth/src/l1_messaging.rs

```diff
@@ -0,0 +1,494 @@
+use anyhow::Context;
+use futures::StreamExt;
+use std::sync::Arc;
+
+use crate::client::StarknetCoreContract::LogMessageToL2;
+use crate::client::{EthereumClient, StarknetCoreContract};
+use crate::utils::u256_to_felt;
+use alloy::primitives::{keccak256, FixedBytes, U256};
+use alloy::sol_types::SolValue;
+use blockifier::transaction::transactions::L1HandlerTransaction as BlockifierL1HandlerTransaction;
+use dc_db::{messaging_db::LastSyncedEventBlock, DeoxysBackend};
+use dp_utils::channel_wait_or_graceful_shutdown;
+use starknet_api::core::{ChainId, ContractAddress, EntryPointSelector, Nonce};
+use starknet_api::transaction::{
+    Calldata, Fee, L1HandlerTransaction, Transaction, TransactionHash, TransactionVersion,
+};
+use starknet_api::transaction_hash::get_transaction_hash;
+use starknet_types_core::felt::Felt;
+
+pub async fn sync(backend: &DeoxysBackend, client: &EthereumClient, chain_id: &ChainId) -> anyhow::Result<()> {
+    tracing::info!("⟠ Starting L1 Messages Syncing...");
+
+    let last_synced_event_block = match backend.messaging_last_synced_l1_block_with_event() {
```

Comment:
event index could be used to uniquely identify that event, it's not used here but could be great to have for monitoring etc

Reply:
But I think you risk reprocessing an event here, if the node shuts down when processing event 2 in a block, you start over from event 1
I thought this event index was there to avoid that